apps/web/utils/digest/index.ts

```diff
@@ -0,0 +1,155 @@
+import { env } from "@/env";
+import { publishToQstashQueue } from "@/utils/upstash";
+import { getCronSecretHeader } from "@/utils/cron";
+import { createScopedLogger } from "@/utils/logger";
+import prisma from "@/utils/prisma";
+import type { DigestBody } from "@/app/api/ai/digest/validation";
+import { DigestStatus } from "@prisma/client";
+import type { DigestEmailSummarySchema } from "@/app/api/resend/digest/validation";
+import type { EmailForAction } from "@/utils/ai/types";
+
+const logger = createScopedLogger("digest");
+
+export async function enqueueDigestItem({
+  email,
+  emailAccountId,
+  actionId,
+  coldEmailId,
+}: {
+  email: EmailForAction;
+  emailAccountId: string;
+  actionId?: string;
+  coldEmailId?: string;
+}) {
+  const url = `${env.NEXT_PUBLIC_BASE_URL}/api/ai/digest`;
+  try {
+    await publishToQstashQueue<DigestBody>({
+      queueName: "digest-item-summarize",
+      parallelism: 3, // Allow up to 3 concurrent jobs from this queue
+      url,
+      body: {
+        emailAccountId,
+        actionId,
+        coldEmailId,
+        message: {
+          id: email.id,
+          threadId: email.threadId,
+          from: email.headers.from,
+          to: email.headers.to,
+          subject: email.headers.subject,
+          content: email.textPlain || "",
+        },
+      },
+      headers: getCronSecretHeader(),
+    });
+  } catch (error) {
+    logger.error("Failed to publish to Qstash", {
+      emailAccountId,
+      error,
+    });
+  }
+}
+
+async function findOldestUnsentDigest(emailAccountId: string) {
+  return prisma.digest.findFirst({
+    where: {
+      emailAccountId,
+      status: DigestStatus.PENDING,
+    },
+    orderBy: {
+      createdAt: "asc",
+    },
+  });
+}
+
+export async function upsertDigest({
```

Comment:
im okay keeping one-time use functions where they're used. and then not exporting. makes it easier to follow an api if it's all in one file